Index: fixincludes/inclhack.def
===================================================================
--- fixincludes/inclhack.def.orig	2011-08-22 05:31:20.303566111 +0000
+++ fixincludes/inclhack.def	2011-08-22 05:31:25.375742362 +0000
@@ -370,6 +370,20 @@
 };
 
 /*
+ * complex.h on AIX 5 and AIX 6 define _Complex_I and I in terms of __I,
+ * which only is provided by AIX xlc C99.
+ */
+fix = {
+    hackname  = aix_complex;
+    mach      = "*-*-aix*";
+    files     = complex.h;
+    select    = "#define[ \t]_Complex_I[ \t]__I";
+    c_fix     = format;
+    c_fix_arg = "#define _Complex_I (__extension__ 1.0iF)";
+    test_text = "#define _Complex_I     __I\n";
+};
+
+/*
  * net/if_arp.h defines a variable fc_softc instead of adding a
  * typedef for the struct on AIX 5.2, 5.3, 6.1 and 7.1
  */
@@ -1872,6 +1886,16 @@
 		"}\n";
 };
 
+fix = {
+    hackname  = hpux_imaginary_i;
+    mach      = "ia64-hp-hpux11.*";
+    files     = complex.h;
+    select    = "^[ \t]*#[ \t]*define[ \t]*_Complex_I.*";
+    c_fix     = format;
+    c_fix_arg = "#define _Complex_I (__extension__ 1.0iF)";
+    test_text = "#define _Complex_I (0.f+_Imaginary_I)\n";
+};
+
 /*
  *  Fix glibc definition of HUGE_VAL in terms of hex floating point constant
  */
@@ -2883,6 +2907,21 @@
 		"#define	I		_Imaginary_I";
 };
 
+/*
+ * Solaris 10+ <complex.h> is wrapped in #ifndef __cplusplus.  Wrap in
+ * extern "C" instead so libstdc++ can use it.
+ */
+fix = {
+    hackname  = solaris_complex_cxx;
+    mach      = "*-*-solaris2.*";
+    files     = complex.h;
+    sed       = "/#if[ \t]*!defined(__cplusplus)/c\\\n"
+                "#ifdef\t__cplusplus\\\nextern \"C\" {\\\n#endif";
+    sed       = "/#endif[ \t]*\\/\\* !defined(__cplusplus) \\*\\//c\\\n"
+                "#ifdef\t__cplusplus\\\n}\\\n#endif";
+    test_text = "#if !defined(__cplusplus)\n"
+                "#endif /* !defined(__cplusplus) */";
+};
 
 /*
  * Solaris 8 PTHREAD_COND_INITIALIZER lacks the __pthread_cond_magic field.
