Index: fixincludes/inclhack.def
===================================================================
--- fixincludes/inclhack.def.orig	2010-12-13 18:22:09.000000000 +0000
+++ fixincludes/inclhack.def	2011-08-22 05:31:20.303566111 +0000
@@ -369,6 +369,19 @@
                 "{...init stuff...}";
 };
 
+/*
+ * net/if_arp.h defines a variable fc_softc instead of adding a
+ * typedef for the struct on AIX 5.2, 5.3, 6.1 and 7.1
+ */
+fix = {
+    hackname  = aix_net_if_arp;
+    mach      = "*-*-aix*";
+    files     = "net/if_arp.h";
+    select    = "^struct  fc_softc \\{";
+    c_fix     = format;
+    c_fix_arg = "typedef struct _fc_softc {";
+    test_text = "struct  fc_softc {";
+};
 
 /*
  *  sys/machine.h on AIX 4.3.3 puts whitespace between a \ and a newline
