Index: src/serv.c
===================================================================
--- src/serv.c.orig	2009-10-19 07:11:29.633447146 +0000
+++ src/serv.c	2009-10-19 07:14:08.537161820 +0000
@@ -47,7 +47,7 @@
 #include "error.h"
 #include "read-file.h"
 
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 /* konqueror cannot handle sending the page in multiple
  * pieces.
Index: src/cli.c
===================================================================
--- src/cli.c.orig	2009-10-19 07:11:29.025852796 +0000
+++ src/cli.c	2009-10-19 07:14:08.760896681 +0000
@@ -43,7 +43,7 @@
 
 #include "common.h"
 #include "cli-gaa.h"
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 #if defined _WIN32 || defined __WIN32__
 int _win_select(int max_fd, fd_set * rfds, fd_set * wfds, fd_set * efds,
Index: src/tls_test.c
===================================================================
--- src/tls_test.c.orig	2009-10-19 07:11:28.644811097 +0000
+++ src/tls_test.c	2009-10-19 07:14:09.161525740 +0000
@@ -31,7 +31,7 @@
 #include <tests.h>
 #include <common.h>
 #include <tls_test-gaa.h>
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 #define SA struct sockaddr
 #define ERR(err,s) if (err==-1) {perror(s);return(1);}
