Index: m4/socklen_t.m4
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ m4/socklen_t.m4	2008-08-20 21:18:41.200896480 +0000
@@ -0,0 +1,40 @@
+dnl Check for socklen_t: historically on BSD it is an int, and in
+dnl POSIX 1g it is a type of its own, but some platforms use different
+dnl types for the argument to getsockopt, getpeername, etc.  So we
+dnl have to test to find something that will work.
+AC_DEFUN([TYPE_SOCKLEN_T],
+[
+   AC_CHECK_TYPE([socklen_t], ,[
+      AC_MSG_CHECKING([for socklen_t equivalent])
+      AC_CACHE_VAL([_cv_socklen_t_equiv],
+      [
+         # Systems have either "struct sockaddr *" or
+         # "void *" as the second argument to getpeername
+         _cv_socklen_t_equiv=
+         for arg2 in "struct sockaddr" void; do
+            for t in int size_t unsigned long "unsigned long"; do
+               AC_TRY_COMPILE([
+                  #include <sys/types.h>
+                  #include <sys/socket.h>
+
+                  int getpeername (int, $arg2 *, $t *);
+               ],[
+                  $t len;
+                  getpeername(0,0,&len);
+               ],[
+                  _cv_socklen_t_equiv="$t"
+                  break
+               ])
+            done
+         done
+
+         if test "x$_cv_socklen_t_equiv" = x; then
+            AC_MSG_ERROR([Cannot find a type to use in place of socklen_t])
+         fi
+      ])
+      AC_MSG_RESULT($_cv_socklen_t_equiv)
+      AC_DEFINE_UNQUOTED(socklen_t, $_cv_socklen_t_equiv,
+			[type to use in place of socklen_t if not defined])],
+      [#include <sys/types.h>
+#include <sys/socket.h>])
+])
Index: configure.in
===================================================================
--- configure.in.orig	2008-06-18 18:31:10.477922479 +0000
+++ configure.in	2008-06-23 16:20:08.041461554 +0000
@@ -57,7 +57,8 @@
 test "x$U" != "x" && AC_MSG_ERROR(Compiler not ANSI compliant)
 
 AC_LIBTOOL_WIN32_DLL
-AM_PROG_LIBTOOL
+AC_LIBTOOL_TAGS()
+AC_PROG_LIBTOOL
 
 dnl
 dnl We process the AC_ARG_WITH first so that later we can modify
@@ -91,12 +92,16 @@
 AC_ARG_WITH(html-dir, AC_HELP_STRING([--with-html-dir=path],
             [path to base html directory, default $datadir/doc/html]),
             [HTML_DIR=$withval], [HTML_DIR='$(datadir)/doc'])
-
-AC_ARG_WITH(html-subdir, AC_HELP_STRING([--with-html-subdir=path],
-            [directory used under html-dir, default $PACKAGE-$VERSION/html]),
-            [test "x$withval" != "x" && HTML_DIR="$HTML_DIR/$withval"],
-            [HTML_DIR="$HTML_DIR/\$(PACKAGE)-\$(VERSION)/html"])
-AC_SUBST(HTML_DIR)
+AC_ARG_WITH(xml-catalog,
+            [AS_HELP_STRING([--with-xml-catalog=@<:@path@:>@],
+            [set XML catalog path to @<:@path@:>@])],[
+            AC_DEFINE_UNQUOTED(XML_XML_DEFAULT_CATALOG, "file://$withval",
+                               [Path to default XML catalog])])
+AC_ARG_WITH(sgml-catalog,
+            [AS_HELP_STRING([--with-sgml-catalog=@<:@path@:>@],
+            [set SGML catalog path to @<:@path@:>@])],[
+            AC_DEFINE_UNQUOTED(XML_SGML_DEFAULT_CATALOG, "$withval",
+                               [Path to default SGML catalog])])
 AC_ARG_WITH(http,
 [  --with-http             add the HTTP support (on)])
 AC_ARG_WITH(iconv,
@@ -345,11 +350,6 @@
 	    if test "x${Z_DIR}" != "x"; then
 		Z_CFLAGS="-I${Z_DIR}/include"
 		Z_LIBS="-L${Z_DIR}/lib -lz"
-		[case ${host} in
-		    *-*-solaris*)
-			Z_LIBS="-L${Z_DIR}/lib -R${Z_DIR}/lib -lz"
-			;;
-		esac]
 	    else
 		Z_LIBS="-lz"
 	    fi]))
@@ -439,8 +439,7 @@
 AC_CHECK_FUNCS(stat _stat signal)
 
 dnl Checking the standard string functions availability
-AC_CHECK_FUNCS(printf sprintf fprintf snprintf vfprintf vsprintf vsnprintf sscanf,,
-               NEED_TRIO=1)
+AC_CHECK_FUNCS(snprintf, , AC_CHECK_LIB(trio, trio_snprintf))
 
 dnl Checking for va_copy availability
 AC_MSG_CHECKING([for va_copy])
@@ -468,32 +467,7 @@
 AC_SEARCH_LIBS(setsockopt, [socket net])
 AC_SEARCH_LIBS(connect, [inet])
 
-dnl Determine what socket length (socklen_t) data type is
-AC_MSG_CHECKING([for type of socket length (socklen_t)])
-AC_TRY_COMPILE2([
-#include <stddef.h>
-#include <sys/types.h>
-#include <sys/socket.h>],[
-(void)getsockopt (1, 1, 1, NULL, (socklen_t *)NULL)],[
-  AC_MSG_RESULT(socklen_t *)
-  XML_SOCKLEN_T=socklen_t],[
-  AC_TRY_COMPILE2([
-#include <stddef.h>
-#include <sys/types.h>
-#include <sys/socket.h>],[
-(void)getsockopt (1, 1, 1, NULL, (size_t *)NULL)],[
-    AC_MSG_RESULT(size_t *)
-    XML_SOCKLEN_T=size_t],[
-    AC_TRY_COMPILE2([
-#include <stddef.h>
-#include <sys/types.h>
-#include <sys/socket.h>],[
-(void)getsockopt (1, 1, 1, NULL, (int *)NULL)],[
-      AC_MSG_RESULT(int *)
-      XML_SOCKLEN_T=int],[
-      AC_MSG_WARN(could not determine)
-      XML_SOCKLEN_T="int"])])])
-AC_DEFINE_UNQUOTED(XML_SOCKLEN_T, $XML_SOCKLEN_T, [Determine what socket length (socklen_t) data type is])
+TYPE_SOCKLEN_T
 
 dnl ***********************Checking for availability of IPv6*******************
 
@@ -587,73 +561,14 @@
   [AC_DEFINE([HAVE_ISINF], [], [Define if isinf is there])]))
 
 XML_LIBDIR='-L${libdir}'
-XML_INCLUDEDIR='-I${includedir}/libxml2'
+XML_INCLUDEDIR='-I${includedir}'
 
 dnl
 dnl Extra flags
 dnl
-XML_CFLAGS=""
 RDL_LIBS=""
 
 dnl
-dnl Workaround for native compilers
-dnl  HP  : http://bugs.gnome.org/db/31/3163.html
-dnl  DEC : Enable NaN/Inf
-dnl
-if test "${GCC}" != "yes" ; then
-    case "${host}" in
-          hppa*-*-hpux* )
-	       CFLAGS="${CFLAGS} -Wp,-H30000"
-	       ;;
-          *-dec-osf* )
-               CFLAGS="${CFLAGS} -ieee"
-               ;;
-	  alpha*-*-linux* )
-	       CFLAGS="${CFLAGS} -ieee"
-	       ;;
-    esac
-else
-    if test "$with_fexceptions" = "yes"
-    then
-        #
-	# Not activated by default because this inflates the code size
-	# Used to allow propagation of C++ exceptions through the library
-	#
-	CFLAGS="${CFLAGS} -fexceptions"
-    fi
-       
-    CFLAGS="${CFLAGS} -pedantic -W -Wformat -Wunused -Wimplicit -Wreturn-type -Wswitch -Wcomment -Wtrigraphs -Wformat -Wchar-subscripts -Wuninitialized -Wparentheses -Wshadow -Wpointer-arith -Wcast-align -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Winline -Wredundant-decls" 
-    case "${host}" in
-          alpha*-*-linux* )
-	       CFLAGS="${CFLAGS} -mieee"
-	       ;;
-	  alpha*-*-osf* )
-	       CFLAGS="${CFLAGS} -mieee"
-	       ;;
-    esac
-fi
-case ${host} in
-    *-*-solaris*)
-        XML_LIBDIR="${XML_LIBDIR} -R${libdir}"
-        ;;
-    hppa*-hp-mpeix)
-        NEED_TRIO=1
-	;;
-    *-*-mingw* | *-*-cygwin* | *-*-msvc* )
-        # If the host is Windows, and shared libraries are disabled, we
-        # need to add -DLIBXML_STATIC to CFLAGS in order for linking to
-        # work properly (without it, xmlexports.h would force the use of
-        # DLL imports, which obviously aren't present in a static
-        # library).
-        if test "x$enable_shared" = "xno"; then
-            XML_CFLAGS="$XML_CFLAGS -DLIBXML_STATIC"
-            CFLAGS="$CFLAGS -DLIBXML_STATIC"
-        fi
-        ;;
-esac
-
-
-dnl
 dnl check for python
 dnl
 
@@ -765,7 +680,7 @@
  
   if test "${libxml_have_dlopen}" = "yes"; then
     case "${host}" in
-      *-*-hpux* )
+      hppa*-*-hpux* )
 	MODULE_EXTENSION=".sl"
 	;;
       * )
@@ -789,53 +704,10 @@
 AC_SUBST(MODULE_EXTENSION)
 AC_SUBST(TEST_MODULES)
 
-dnl
-dnl Tester makes use of readline if present
-dnl
-
-dnl
-dnl specific tests to setup DV and Bill's devel environments with debug etc ...
-dnl (-Wunreachable-code)
-dnl
-if [[ "${LOGNAME}" = "veillard" -a "`pwd`" = "/u/veillard/XML" ]] || \
-   [[ "${LOGNAME}" = "veillard" -a "`pwd`" = "/home/veillard/libxml2" ]] || \
-   [[ "${LOGNAME}" = "bill" -a "`pwd`" = "/home/bill/gnomesvn/libxml2" ]]
-   then
-    if test "$with_minimum" != "yes"
-    then
-	if test "${with_mem_debug}" = "" ; then
-	    echo Activating memory debugging
-	    with_mem_debug="yes"
-	    with_run_debug="yes"
-	fi
-	if test "${with_docbook}" = "" ; then
-	    with_docbook="yes"
-	fi
-    fi
-    if test "${GCC}" = "yes" ; then
-    CFLAGS="-g -O -pedantic -W -Wformat -Wunused -Wimplicit -Wreturn-type -Wswitch -Wcomment -Wtrigraphs -Wformat -Wchar-subscripts -Wuninitialized -Wparentheses -Wshadow -Wpointer-arith -Wcast-align -Wwrite-strings -Waggregate-return -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Winline -Wredundant-decls -Wall"
-    fi
-    STATIC_BINARIES="-static"
-dnl -Wcast-qual -ansi
-else
-    STATIC_BINARIES=
-fi
+STATIC_BINARIES=
 AC_SUBST(STATIC_BINARIES)
 
 dnl
-dnl Check for trio string functions
-dnl
-
-if test "${NEED_TRIO}" = "1" ; then
-    echo Adding trio library for string functions
-    WITH_TRIO=1
-else    
-    WITH_TRIO=0
-fi
-AM_CONDITIONAL(WITH_TRIO_SOURCES, test "${NEED_TRIO}" = "1")
-AC_SUBST(WITH_TRIO)
-
-dnl
 dnl Allow to enable/disable various pieces
 dnl
 echo Checking configuration requirements
@@ -855,12 +727,24 @@
 else
     echo Enabling multithreaded support
 
-    AC_CHECK_HEADER(pthread.h,
-	AC_CHECK_LIB(pthread, pthread_join,[
-	   THREAD_LIBS="-lpthread"
-	   AC_DEFINE([HAVE_LIBPTHREAD], [], [Define if pthread library is there (-lpthread)])
-	   AC_DEFINE([HAVE_PTHREAD_H], [], [Define if <pthread.h> is there])
-	   WITH_THREADS="1"]))
+    AC_CHECK_HEADER(pthread.h,[
+        AC_MSG_CHECKING([for pthread_join in -lpthread])
+        save_LIBS=$LIBS
+        LIBS="$LIBS -lpthread"
+        AC_TRY_LINK([#include <pthread.h>],[
+pthread_t pth;
+void *value;
+
+pthread_join(pth,&value);],[
+           AC_MSG_RESULT([yes])
+           THREAD_LIBS="-lpthread"
+           AC_DEFINE([HAVE_LIBPTHREAD], ,
+                     [Define if pthread library is there (-lpthread)])
+           AC_DEFINE([HAVE_PTHREAD_H], ,
+                     [Define if <pthread.h> is there])
+           WITH_THREADS="1"],[
+           AC_MSG_RESULT([no])
+           LIBS=$save_LIBS])])
     case $host_os in
        *mingw32*) WITH_THREADS="1"
            THREADS_W32="Win32"
@@ -909,7 +793,7 @@
 AC_SUBST(THREAD_CFLAGS)
 AC_SUBST(TEST_THREADS)
 AC_SUBST(THREADS_W32)
-
+AM_CONDITIONAL([USE_THREADS],[test "$WITH_THREADS" = "1"])
 dnl
 dnl xmllint shell history
 dnl
@@ -1371,10 +1255,6 @@
 AC_SUBST(CYGWIN_EXTRA_LDFLAGS)
 AC_SUBST(CYGWIN_EXTRA_PYTHON_LIBADD)
 
-AC_SUBST(CPPFLAGS)
-AC_SUBST(CFLAGS)
-AC_SUBST(XML_CFLAGS)
-
 AC_SUBST(XML_LIBDIR)
 AC_SUBST(XML_LIBS)
 AC_SUBST(XML_LIBTOOLLIBS)
Index: doc/examples/Makefile.am
===================================================================
--- doc/examples/Makefile.am.orig	2008-06-18 18:31:08.077787391 +0000
+++ doc/examples/Makefile.am	2008-06-18 18:33:09.645553065 +0000
@@ -1,3 +1,5 @@
+EXAMPLES_DIR = $(HTML_DIR)/examples
+
 # Beware this is autogenerated by index.py
 INCLUDES = -I$(top_builddir)/include -I$(top_srcdir)/include -I@srcdir@/include @THREAD_CFLAGS@ @Z_CFLAGS@
 DEPS = $(top_builddir)/libxml2.la
@@ -12,8 +14,8 @@
 	-@(xsltproc examples.xsl examples.xml && echo "Rebuilt web page" && xmllint --valid --noout index.html)
 
 install-data-local: 
-	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)
-	-@INSTALL@ -m 0644 $(srcdir)/*.html $(srcdir)/*.c $(srcdir)/*.xml $(srcdir)/*.xsl $(srcdir)/*.res $(DESTDIR)$(HTML_DIR)
+	$(mkinstalldirs) $(DESTDIR)$(EXAMPLES_DIR)
+	-@INSTALL@ -m 0644 $(srcdir)/*.html $(srcdir)/*.c $(srcdir)/*.xml $(srcdir)/*.xsl $(srcdir)/*.res $(DESTDIR)$(EXAMPLES_DIR)
 
 EXTRA_DIST=examples.xsl index.py examples.xml writer.xml test3.xml tst.xml test1.xml test2.xml io2.res xpath2.res tree1.res reader4.res io1.res tree2.res reader3.res xpath1.res reader1.res 
 
Index: include/libxml/Makefile.am
===================================================================
--- include/libxml/Makefile.am.orig	2008-06-18 18:31:10.439651592 +0000
+++ include/libxml/Makefile.am	2008-06-18 18:33:09.655990771 +0000
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-xmlincdir = $(includedir)/libxml2/libxml
+xmlincdir = $(includedir)/libxml
 
 xmlinc_HEADERS = \
 		SAX.h \
Index: libxml-2.0.pc.in
===================================================================
--- libxml-2.0.pc.in.orig	2008-06-18 18:31:10.203796921 +0000
+++ libxml-2.0.pc.in	2008-06-18 18:33:09.669588301 +0000
@@ -7,7 +7,7 @@
 Name: libXML
 Version: @VERSION@
 Description: libXML library version2.
-Requires:
+Requires: libiconv
 Libs: -L${libdir} -lxml2
 Libs.private: @THREAD_LIBS@ @Z_LIBS@ @ICONV_LIBS@ @M_LIBS@ @LIBS@
-Cflags: @XML_INCLUDEDIR@ @XML_CFLAGS@
+Cflags: -I${includedir}
Index: python/Makefile.am
===================================================================
--- python/Makefile.am.orig	2008-06-18 18:31:09.448407282 +0000
+++ python/Makefile.am	2008-06-18 19:34:21.825694450 +0000
@@ -9,7 +9,7 @@
 	-I$(top_builddir)/include \
 	-I$(top_builddir)/$(subdir)
 
-docsdir = $(datadir)/doc/libxml2-python-$(LIBXML_VERSION)
+docsdir = $(HTML_DIR)
 # libxml2class.txt is generated
 dist_docs_DATA = TODO
 
@@ -25,7 +25,7 @@
 
 if WITH_PYTHON
 mylibs = \
-	$(top_builddir)/libxml2.la
+	$(libdir)/libxml2.la
 
 python_LTLIBRARIES = libxml2mod.la
 
Index: python/tests/Makefile.am
===================================================================
--- python/tests/Makefile.am.orig	2008-06-18 18:31:09.372662802 +0000
+++ python/tests/Makefile.am	2008-06-23 16:46:29.462014960 +0000
@@ -1,4 +1,4 @@
-exampledir = $(datadir)/doc/libxml2-python-$(LIBXML_VERSION)/examples
+exampledir = $(HTML_DIR)/examples
 dist_example_DATA = $(PYTESTS) $(XMLS)
 
 PYTESTS=	\
@@ -36,7 +36,6 @@
     readererr.py\
     relaxng.py	\
     schema.py	\
-    thread2.py \
     sync.py \
     tstLastError.py \
     indexes.py \
@@ -48,6 +47,10 @@
     compareNodes.py \
     xpathns.py
 
+if USE_THREADS
+PYTESTS += thread2.py
+endif
+
 XMLS=		\
     tst.xml	\
     valid.xml	\
Index: xml2-config.in
===================================================================
--- xml2-config.in.orig	2008-06-18 18:31:07.638879728 +0000
+++ xml2-config.in	2008-08-20 21:19:47.545938205 +0000
@@ -67,7 +67,7 @@
 	;;
 
     --cflags)
-       	echo @XML_INCLUDEDIR@ @XML_CFLAGS@
+       	echo @XML_INCLUDEDIR@
        	;;
 
     --libtool-libs)
@@ -82,17 +82,7 @@
        	;;
 
     --libs)
-        if [ "`uname`" = "Linux" ]
-	then
-	    if [ "@XML_LIBDIR@" = "-L/usr/lib" -o "@XML_LIBDIR@" = "-L/usr/lib64" ]
-	    then
-		echo @XML_LIBS@ 
-	    else
-		echo @XML_LIBDIR@ @XML_LIBS@ 
-	    fi
-	else
-	    echo @XML_LIBDIR@ @XML_LIBS@ @WIN32_EXTRA_LIBADD@
-	fi
+	echo -L${libdir} -lxml2
        	;;
 
     *)
Index: Makefile.am
===================================================================
--- Makefile.am.orig	2008-06-18 18:31:10.460753576 +0000
+++ Makefile.am	2008-06-18 18:33:09.916334693 +0000
@@ -19,17 +19,6 @@
 
 libxml2_la_LDFLAGS = @CYGWIN_EXTRA_LDFLAGS@ @WIN32_EXTRA_LDFLAGS@ -version-info @LIBXML_VERSION_INFO@ @MODULE_PLATFORM_LIBS@
 
-if WITH_TRIO_SOURCES
-libxml2_la_SOURCES = SAX.c entities.c encoding.c error.c parserInternals.c  \
-		parser.c tree.c hash.c list.c xmlIO.c xmlmemory.c uri.c  \
-		valid.c xlink.c HTMLparser.c HTMLtree.c debugXML.c xpath.c  \
-		xpointer.c xinclude.c nanohttp.c nanoftp.c DOCBparser.c \
-		catalog.c globals.c threads.c c14n.c xmlstring.c \
-		xmlregexp.c xmlschemas.c xmlschemastypes.c xmlunicode.c \
-		triostr.c trio.c xmlreader.c relaxng.c dict.c SAX2.c \
-		xmlwriter.c legacy.c chvalid.c pattern.c xmlsave.c \
-		xmlmodule.c schematron.c
-else
 libxml2_la_SOURCES = SAX.c entities.c encoding.c error.c parserInternals.c  \
 		parser.c tree.c hash.c list.c xmlIO.c xmlmemory.c uri.c  \
 		valid.c xlink.c HTMLparser.c HTMLtree.c debugXML.c xpath.c  \
@@ -39,7 +28,6 @@
 		xmlreader.c relaxng.c dict.c SAX2.c \
 		xmlwriter.c legacy.c chvalid.c pattern.c xmlsave.c \
 		xmlmodule.c schematron.c
-endif
 
 DEPS = $(top_builddir)/libxml2.la
 LDADDS = @STATIC_BINARIES@ $(top_builddir)/libxml2.la @THREAD_LIBS@ @Z_LIBS@ $(ICONV_LIBS) @M_LIBS@ @WIN32_EXTRA_LIBADD@
@@ -1166,11 +1154,11 @@
 #
 BASE_DIR=$(datadir)/doc
 DOC_MODULE=libxml2-$(VERSION)
-EXAMPLES_DIR=$(BASE_DIR)/$(DOC_MODULE)/examples
+EXAMPLES_DIR=$(HTML_DIR)/examples
 
 install-data-local: 
-	$(mkinstalldirs) $(DESTDIR)$(BASE_DIR)/$(DOC_MODULE)
-	-@INSTALL@ -m 0644 $(srcdir)/Copyright $(DESTDIR)$(BASE_DIR)/$(DOC_MODULE)
+	$(mkinstalldirs) $(DESTDIR)$(HTML_DIR)
+	-@INSTALL@ -m 0644 $(srcdir)/Copyright $(DESTDIR)$(HTML_DIR)
 	$(mkinstalldirs) $(DESTDIR)$(EXAMPLES_DIR)
 	-@INSTALL@ -m 0644 $(srcdir)/xmllint.c $(DESTDIR)$(EXAMPLES_DIR)
 	-@INSTALL@ -m 0644 $(srcdir)/testSAX.c $(DESTDIR)$(EXAMPLES_DIR)
Index: include/libxml/xmlversion.h.in
===================================================================
--- include/libxml/xmlversion.h.in.orig	2007-05-28 14:08:51.000000000 +0000
+++ include/libxml/xmlversion.h.in	2008-06-18 18:33:09.924333273 +0000
@@ -60,31 +60,6 @@
  */
 #define LIBXML_TEST_VERSION xmlCheckVersion(@LIBXML_VERSION_NUMBER@);
 
-#ifndef VMS
-#if @WITH_TRIO@
-/**
- * WITH_TRIO:
- *
- * defined if the trio support need to be configured in
- */
-#define WITH_TRIO
-#else
-/**
- * WITHOUT_TRIO:
- *
- * defined if the trio support should not be configured in
- */
-#define WITHOUT_TRIO
-#endif
-#else /* VMS */
-/**
- * WITH_TRIO:
- *
- * defined if the trio support need to be configured in
- */
-#define WITH_TRIO 1
-#endif /* VMS */
-
 /**
  * LIBXML_THREAD_ENABLED:
  *
Index: libxml.h
===================================================================
--- libxml.h.orig	2007-11-23 10:47:23.000000000 +0000
+++ libxml.h	2008-06-18 18:33:09.944849118 +0000
@@ -40,7 +40,7 @@
 int vfprintf(FILE *, const char *, va_list);
 #endif
 
-#ifndef WITH_TRIO
+#ifdef HAVE_SNPRINTF
 #include <stdio.h>
 #else
 /**
Index: python/libxml.c
===================================================================
--- python/libxml.c.orig	2007-01-09 21:22:52.000000000 +0000
+++ python/libxml.c	2008-06-18 18:33:09.957293936 +0000
@@ -28,7 +28,7 @@
 
 #if (defined(_MSC_VER) || defined(__MINGW32__)) && !defined(vsnprintf)
 #define vsnprintf(b,c,f,a) _vsnprintf(b,c,f,a)
-#elif defined(WITH_TRIO)
+#elif !defined(HAVE_SNPRINTF)
 #include "trio.h"
 #define vsnprintf trio_vsnprintf
 #endif
