Index: setupinfo.py
===================================================================
--- setupinfo.py.orig	2009-11-11 15:15:46.000000000 +0000
+++ setupinfo.py	2009-11-27 19:53:59.895031596 +0000
@@ -153,7 +153,7 @@
     elif OPTION_STATIC:
         libs = ['z', 'm']
     else:
-        libs = ['xslt', 'exslt', 'xml2', 'z', 'm']
+        libs = ['xslt', 'exslt', 'xml2']
     return libs
 
 def library_dirs(static_library_dirs):
