--- lib/misclib.py	Mon Apr 19 17:09:26 2004
+++ lib/misclib.py.new	Mon Apr 19 17:13:02 2004
@@ -302,6 +302,8 @@
   elif uname[0] == 'IRIX' or uname[0] == 'IRIX64':
     if uname[2] == '6.2': return 'irix', '6', '2'
     if uname[2] == '6.5': return 'irix', '6', '5'
+  elif uname[0] == 'Darwin':
+    if uname[2] == '8.8.0': return 'MacOS', '10', '8'
   elif uname[0] == 'Linux':
     if uname[4] == 'i686': return 'linux', '7', '1'
   elif uname[0] == 'OSF1':
