--- xml/parsers/xmlproc/xmlutils.py.orig	2005-01-30 23:10:52.000000000 -0600
+++ xml/parsers/xmlproc/xmlutils.py	2005-01-30 23:11:01.000000000 -0600
@@ -720,6 +720,7 @@
             # to the recoding.
             try:
                 self.data = self.charset_converter(self.data)
+                self.datasize = len(self.data)
             except UnicodeError, e:
                 self._handle_decoding_error(self.data, e)
             self.input_encoding = enc1
