--- lib/systype	Tue Aug  5 17:57:37 2003
+++ lib/systype.new	Mon Apr 19 17:13:48 2004
@@ -1,9 +1,9 @@
-#! /bin/ksh
+#! /bin/bash
 # Attempt to guess a canonical system name.
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
 #   2000, 2001, 2002, 2003 Free Software Foundation, Inc.
 
-timestamp='2003-02-22'
+timestamp='2004-04-19'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
