Index: m4/include_next.m4
===================================================================
--- m4/include_next.m4.orig	2008-05-14 12:53:25.000000000 +0000
+++ m4/include_next.m4	2008-11-06 15:50:36.527645661 +0000
@@ -15,6 +15,7 @@
      mkdir conftestd1 conftestd2
      cat <<EOF > conftestd1/conftest.h
 #define DEFINED_IN_CONFTESTD1
+#include <stdio.h>
 #include_next <conftest.h>
 #ifdef DEFINED_IN_CONFTESTD2
 int foo;
