diff --git a/m4/jm-macros.m4 b/m4/jm-macros.m4
index 9680e95..062737b 100644
--- m4/jm-macros.m4
+++ m4/jm-macros.m4
@@ -1,4 +1,4 @@
-#serial 106   -*- autoconf -*-
+#serial 107   -*- autoconf -*-
 
 dnl Misc type-related macros for coreutils.
 
@@ -49,6 +49,9 @@ AC_DEFUN([coreutils_MACROS],
   # Used by install.c.
   AC_CHECK_FUNCS_ONCE([matchpathcon_init_prefix])
 
+  # Used by sort.c.
+  AC_CHECK_FUNCS_ONCE([nl_langinfo])
+
   AC_CHECK_FUNCS_ONCE( \
     endgrent \
     endpwent \
