Index: src/serv.c
===================================================================
--- src/serv.c.orig	2008-10-22 14:58:36.908219089 +0000
+++ src/serv.c	2008-10-22 15:00:19.024911006 +0000
@@ -47,7 +47,7 @@
 #include "error.h"
 #include "read-file.h"
 
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 /* konqueror cannot handle sending the page in multiple
  * pieces.
Index: src/cli.c
===================================================================
--- src/cli.c.orig	2008-10-22 08:42:02.000000000 +0000
+++ src/cli.c	2008-10-22 15:00:08.980353427 +0000
@@ -43,7 +43,7 @@
 
 #include "common.h"
 #include "cli-gaa.h"
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 #if defined _WIN32 || defined __WIN32__
 int _win_select(int max_fd, fd_set * rfds, fd_set * wfds, fd_set * efds,
Index: src/tls_test.c
===================================================================
--- src/tls_test.c.orig	2008-10-22 08:42:02.000000000 +0000
+++ src/tls_test.c	2008-10-22 15:00:00.792268584 +0000
@@ -31,7 +31,7 @@
 #include <tests.h>
 #include <common.h>
 #include <tls_test-gaa.h>
-#include "getaddrinfo.h"
+#include <netdb.h>
 
 #define SA struct sockaddr
 #define ERR(err,s) if (err==-1) {perror(s);return(1);}
