Index: gzip.c
===================================================================
--- gzip.c.orig	2007-03-20 05:09:51.000000000 +0000
+++ gzip.c	2008-06-26 17:22:58.208103587 +0000
@@ -1273,7 +1273,7 @@
 	}
 	if ((flags & CONTINUATION) != 0) {
 	    fprintf(stderr,
-		    "%s: %s is a a multi-part gzip file -- not supported\n",
+		    "%s: %s is a multi-part gzip file -- not supported\n",
 		    program_name, ifname);
 	    exit_code = ERROR;
 	    if (force <= 1) return -1;
Index: zdiff.in
===================================================================
--- zdiff.in.orig	2007-02-08 22:54:59.000000000 +0000
+++ zdiff.in	2008-06-26 17:22:22.382661026 +0000
@@ -104,7 +104,7 @@
 			  exec 4>&1
 			  (gzip -cdfq -- "$1" 4>&-; echo $? >&4) 3>&- |
 			    ( (gzip -cdfq -- "$2" 4>&-; echo $? >&4) 3>&- 5<&- </dev/null |
-			       eval "$cmp" /dev/fd/5 -) 5<&0
+			       eval "$cmp" /dev/fd/5 - >&3) 5<&0
 			)
 			case $gzip_status in
 			  *[1-9]*) gzip_status=1;;
